### PR TITLE
api: use upstream offered JSON files

### PR DESCRIPTION
**Pull request by aparcar**
This is right now for testing only, however ideally we'd switch to automatically generated JSON files instead of relying on the collect script.

### CODE REVIEW

File: www/index.js [hunk 1]
```diff
@@ -739,10 +742,29 @@ function init() {
     show("#details_custom");
   }
 
-  setupSelectList($("#versions"), Object.keys(config.versions), (version) => {
+  let versions = await fetch(config.image_url + "/.versions.json", { cache: "no-cache" })
+    .then((obj) => { return obj.json(); })
+    .then((obj) => { return obj.versions_list.filter((version) => version.localeCompare("19.07.4") >= 0);});
```
Comment (mwarning) [inline www/index.js:747]: Why is this needed?

File: www/index.js [hunk 2]
```diff
@@ -739,10 +742,29 @@ function init() {
     show("#details_custom");
   }
 
-  setupSelectList($("#versions"), Object.keys(config.versions), (version) => {
+  let versions = await fetch(config.image_url + "/.versions.json", { cache: "no-cache" })
```
Comment (mwarning) [inline www/index.js:745]: What is the point of putting the versions from the `config.json` to a `.versions.json`. I do not understand yet. :-)

File: www/index.js [hunk 3]
```diff
@@ -665,14 +665,17 @@ function setModel(overview, target, id) {
 function changeModel(version, overview, title, base_url) {
   const entry = overview.profiles[title];
   if (entry) {
-    fetch(`${base_url}/${entry.target}/${entry.id}.json`, {
+    fetch(`${base_url}/targets/${entry.target}/profiles.json`, {
```
Comment (mwarning) [inline www/index.js:668]: It seems that instead of the individual JSON files, the OpenWrt generated profiles.json files are used. Those file contain all models of the target. This OK.

But we would still need to create the overview.json file. No?

File: www/index.js [hunk 4]
```diff
@@ -739,10 +742,29 @@ function init() {
...
 
-  setupSelectList($("#versions"), Object.keys(config.versions), (version) => {
+  let versions = await fetch(config.image_url + "/.versions.json", { cache: "no-cache" })
+    .then((obj) => { return obj.json(); })
+    .then((obj) => { return obj.versions_list.filter((version) => version.localeCompare("19.07.4") >= 0);});
+
+  if (config.show_snapshots) {
+    let branches = [];
+    for (const version of versions.slice()) {
+      let branch = version.split('.').slice(0, -1).join('.') + '-SNAPSHOT';
```
Comment (mwarning) [inline www/index.js:752]: I do not like to hard code this OpenWrt specific stuff. I wonder if this can be done in a nicer way.
Shouldn't we but this in the .versions.json file?

File: www/index.js [hunk 5]
```diff
@@ -739,10 +742,29 @@ function init() {
...
+    }
+    versions.push("SNAPSHOTS");
+  }
+
+  setupSelectList($("#versions"), versions, (version) => {
     // A new version was selected
-    let base_url = config.versions[version];
-    fetch(base_url + "/overview.json", { cache: "no-cache" })
+    let base_url = `${config.image_url}/releases/${version}`;
+    if (version == "SNAPSHOTS") {
```
Comment (mwarning) [inline www/index.js:763]: This is quiet a hack. Can we avoid to make this project too OpenWrt specific?

File: www/index.js (hunk 3 above)
Comment (aparcar) [reply www/index.js:668]: It's now outomatocally created

File: www/index.js (hunk 4 above)
Comment (aparcar) [reply www/index.js:752]: Sure would would you add it?

File: www/index.js (hunk 5 above)
Comment (aparcar) [reply www/index.js:763]: Sure, what do you suggest?

File: www/index.js (hunk 3 above)
Comment (mwarning) [reply www/index.js:668]: I am not sure what  automatic means. Is the overview.json now generated by the OpenWrt build process?

Comment (aparcar) [reply www/index.js:668]: Correct

File: www/index.js (hunk 1 above)
Comment (aparcar) [reply www/index.js:747]: Versions prior to 19.07.4 wouldn't offer any JSON files so rendering the selector useless.

File: www/index.js (hunk 2 above)
Comment (aparcar) [reply www/index.js:745]: .versions.json is automatically created be the downloads.openwrt.org server while the config requires an external service. The idea is to have firmware-selector as a website hosted that doesn't need extra scripts to run

**Comment by mwarning** [comment]
@aparcar these changes make things simpler. But I haven't found the concentration to make this change elegant yet.

File: www/index.js (hunk 3 above)
Comment (mwarning) [reply www/index.js:668]: Is that code now part of https://github.com/openwrt/openwrt ?

**Comment by aparcar** [comment]
Please take your time

File: www/index.js (hunk 3 above)
Comment (aparcar) [reply www/index.js:668]: No it's part of the way we deploy downloads.openwrt.org
